### ISSUE TITLE
Don't get mouse cursor when started from tty

### ISSUE THREAD

**Issue by thallium** [taiwins/taiwins#217]
Got error: EGL extension EGL_KHR-swap_buffers_with_damage not found

**Comment by xeechou** [comment]
Hi, I don't think that EGL extension is related to cursor. 

If you are using the taiwins shell, you need to trigger a `wl_pointer.enter` event by any pointer event, for example, moving the cursor.

Regards

**Comment by thallium** [comment]
I tried but still can't can't get it.

**Comment by xeechou** [comment]
Hi, could you get the Wayland debug logging?

You can enable it through
 

   export WAYLAND_DEBUG=1 /path/to/taiwins 2> /path/to/log

**Comment by thallium** [comment]
Here it is: https://paste.ubuntu.com/p/dMG9RzpFY7/

**Comment by xeechou** [comment]
from your log you can see that the `wl_seat@8.capabilities(2)`, which means it does not have a pointer device. 

There should be a `wl_seat@8.get_pointer` which there isn't. That is why you don't see a cursor. I think somehow your mouse is not recognized. You can try to plug another one to see if it works

**Comment by thallium** [comment]
Yes you are correct, I tried another mouse and it worked. So is there a way to let my Logitech mouse registered as a mouse?

**Comment by xeechou** [comment]
This is likely  specific to your Logitech mouse. But anyway, congrats, you found a new bug in taiwins. Unfortunately it's not easy to fix.

Reason is that libinput advertise your mouse as both keyboard and a pointer. However, I didn't consider the cases of Multi-purpose input device. For now I don't see a good solution to support the multi-purpose input devices, maybe creating multiple input devices separately? It seems complicated to manage
